src/cljs/main/broadfcui/common/table.cljs

```diff
@@ -211,14 +211,14 @@
      (update restored :column-meta
              (fn [cols]
                (let [headers-restored (set (map #(or (:header-key %) (:header %)) cols))
-                     col-headers (map #(or (:header-key %) (:header %)) (:columns props))
+                     col-headers (set (map #(or (:header-key %) (:header %)) (:columns props)))
                      unmentioned-headers (set (remove #(contains? headers-restored %) col-headers))
                      unmentioned-cols (->> (:columns props)
                                            (filter #(contains? unmentioned-headers (:header %)))
                                            (map #(select-keys % [:header]))
-                                           (map #(assoc % :width 100 :visible? true)))]
-
-                 (vec (concat cols unmentioned-cols))))))))
+                                           (map #(assoc % :width 100 :visible? true)))
+                     extant-cols (filterv #(contains? col-headers (or (:header-key %) (:header %))) cols)]
+                 (vec (concat extant-cols unmentioned-cols))))))))
```

Comment:
Maybe a little easier to follow (not tested)?
```cljs
(fn [restored-column-meta]
  (let [get-header #(or (:header-key %) (:header %))
        specified-headers (set (map get-header (:columns props)))
        ;; Restored columns that are not specified are from an old version of the persisted columns.
        restored-column-meta (filter #(contains? specified-headers (get-header %)) restored-column-meta)
        restored-headers (set (map get-header restored-column-meta))
        new-headers (clojure.set/difference specified-headers restored-headers)
        new-cols (->> (:columns props) (filter #(contains? new-headers (get-header %)))
              (map #(select-keys % [:header])) (map #(assoc % :width 100 :visible? true))]
    (vec (concat restored-column-meta new-cols)))
```
Also, why does `select-keys` not also grab `:header-key`?